Source/Core/DolphinQt2/Resources.cpp

```diff
@@ -13,25 +17,37 @@ QList<QPixmap> Resources::m_countries;
 QList<QPixmap> Resources::m_ratings;
 QList<QPixmap> Resources::m_misc;
 
+QString Resources::GetPreferredIconScale(int max)
+{
+  const auto dpr = QGuiApplication::primaryScreen()->devicePixelRatio();
+  const auto scale = std::min(dpr <= 1 ? 1 : (dpr <= 2) ? 2 : 4, max);
+
+  if (scale > 1)
+    return QStringLiteral("@%1x").arg(scale);
+  else
+    return QStringLiteral("");
+}
+
 void Resources::Init()
 {
   QString sys_dir = QString::fromStdString(File::GetSysDirectory() + RESOURCES_DIR + DIR_SEP);
 
-  QStringList platforms{QStringLiteral("Platform_Gamecube.png"), QStringLiteral("Platform_Wii.png"),
-                        QStringLiteral("Platform_Wad.png"), QStringLiteral("Platform_File.png")};
+  QStringList platforms{QStringLiteral("Platform_Gamecube%1.png"),
+                        QStringLiteral("Platform_Wii%1.png"), QStringLiteral("Platform_Wad%1.png"),
+                        QStringLiteral("Platform_File%1.png")};
   for (QString platform : platforms)
-    m_platforms.append(QPixmap(platform.prepend(sys_dir)));
+    m_platforms.append(QPixmap(platform.arg(GetPreferredIconScale()).prepend(sys_dir)));
 
   QStringList countries{
-      QStringLiteral("Flag_Europe.png"),        QStringLiteral("Flag_Japan.png"),
-      QStringLiteral("Flag_USA.png"),           QStringLiteral("Flag_Australia.png"),
-      QStringLiteral("Flag_France.png"),        QStringLiteral("Flag_Germany.png"),
-      QStringLiteral("Flag_Italy.png"),         QStringLiteral("Flag_Korea.png"),
-      QStringLiteral("Flag_Netherlands.png"),   QStringLiteral("Flag_Russia.png"),
-      QStringLiteral("Flag_Spain.png"),         QStringLiteral("Flag_Taiwan.png"),
-      QStringLiteral("Flag_International.png"), QStringLiteral("Flag_Unknown.png")};
+      QStringLiteral("Flag_Europe%1.png"),        QStringLiteral("Flag_Japan%1.png"),
+      QStringLiteral("Flag_USA%1.png"),           QStringLiteral("Flag_Australia%1.png"),
+      QStringLiteral("Flag_France%1.png"),        QStringLiteral("Flag_Germany%1.png"),
+      QStringLiteral("Flag_Italy%1.png"),         QStringLiteral("Flag_Korea%1.png"),
+      QStringLiteral("Flag_Netherlands%1.png"),   QStringLiteral("Flag_Russia%1.png"),
+      QStringLiteral("Flag_Spain%1.png"),         QStringLiteral("Flag_Taiwan%1.png"),
+      QStringLiteral("Flag_International%1.png"), QStringLiteral("Flag_Unknown%1.png")};
   for (QString country : countries)
-    m_countries.append(QPixmap(country.prepend(sys_dir)));
+    m_countries.append(QPixmap(country.arg(GetPreferredIconScale()).prepend(sys_dir)));
```

Comment:
This copy-pasting is getting unwieldy, and its common logic should be extracted out. Depending on how its done, the QStringLists probably aren't even needed, it could potentially just be

```c++
m_countries.append(LoadDolphinPixmap("Flag_Europe"));
m_countries.append(LoadDolphinPixmap("Flag_Japan"));
```

etc...